src/reducer/product-reducer.js

```diff
@@ -0,0 +1,90 @@
+const ProductReducer = (state, action) => {
+	switch (action.type) {
+		case "APPLE":
+			return {
+				...state,
+				categories: {
+					...state["categories"],
+					Apple: !state.categories.Apple,
+				},
+			};
+		case "SAMSUNG":
+			return {
+				...state,
+				categories: {
+					...state["categories"],
+					Samsung: !state.categories.Samsung,
+				},
+			};
+		case "REDMI":
+			return {
+				...state,
+				categories: {
+					...state["categories"],
+					Redmi: !state.categories.Redmi,
+				},
+			};
+		case "REALME":
+			return {
+				...state,
+				categories: {
+					...state["categories"],
+					Realme: !state.categories.Realme,
+				},
+			};
+		case "ONEPLUS":
+			return {
+				...state,
+				categories: {
+					...state["categories"],
+					Oneplus: !state.categories.Oneplus,
+				},
+			};
+		case "OPPO":
+			return {
+				...state,
+				categories: {
+					...state["categories"],
```

Comment:
`...state.categories` will work as well but since its working, no need to make changes here